src/MyHealth/DayIndicator.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { FunctionComponent } from "react"
+import { View, Text, ViewStyle, TextStyle, StyleSheet } from "react-native"
+import dayjs from "dayjs"
+
+import { DateTimeUtils } from "../utils"
+import { Outlines, Colors, Typography, Spacing } from "../styles"
+import { CheckInStatus, DayLogData } from "./symptoms"
+
+interface DayIndicatorProps {
+  logData: DayLogData
+  isSelected: boolean
+}
+
+type IndicatorStyle = [ViewStyle, TextStyle]
+
+const DayIndicator: FunctionComponent<DayIndicatorProps> = ({
+  logData,
+  isSelected,
+}: DayIndicatorProps) => {
+  const isToday = DateTimeUtils.isToday(logData.date)
+
+  const applyRiskStyle = ([
+    circleStyle,
+    textStyle,
+  ]: IndicatorStyle): IndicatorStyle => {
+    switch (logData.checkIn.status) {
+      case CheckInStatus.NotCheckedIn: {
+        return [{ ...circleStyle }, { ...textStyle, color: Colors.primaryText }]
+      }
+      case CheckInStatus.TooOld: {
+        return [
+          { ...circleStyle },
+          { ...textStyle, color: Colors.transparentPrimaryText },
+        ]
+      }
+      case CheckInStatus.FeelingGood: {
+        return [
+          { ...circleStyle, borderColor: Colors.success100 },
+          { ...textStyle, color: Colors.success100 },
+        ]
+      }
+      case CheckInStatus.FeelingNotWell: {
+        return [
+          { ...circleStyle, borderColor: Colors.warning100 },
+          { ...textStyle, color: Colors.warning100 },
+        ]
+      }
+    }
+  }
+
+  const applyIsTodayStyle = ([
+    circleStyle,
+    textStyle,
+  ]: IndicatorStyle): IndicatorStyle => {
+    if (isToday) {
+      return [
+        {
+          ...circleStyle,
+        },
+        {
+          ...textStyle,
+          fontWeight: Typography.boldWeight,
+        },
+      ]
+    } else {
+      return [circleStyle, textStyle]
+    }
+  }
+
+  const applyIsSelectedStyle = ([
+    circleStyle,
+    textStyle,
+  ]: IndicatorStyle): IndicatorStyle => {
+    if (isSelected) {
+      switch (logData.checkIn.status) {
```

Comment:
Why do we need to switch on the checkIn.status twice? can we consolidate the conditional logic here?